### PR TITLE
FEATURE: add user opt to select bookmarks as home

### PR DESCRIPTION
**Pull request by merefield**
Admins can currently add the bookmarks discovery route link
to the homepage interface, but users can't presently select
that as their default home view.  This change facilitates that,
adding the option to the existing Default Home Page dropdown on
the User Preferences Interface page.

### CODE REVIEW

**Comment by eviltrout** [comment]
Hi @merefield could you rebase on top of master? There is linting failures that I think were fixed in the latest release and it's making it hard to review without them.

**Comment by merefield** [comment]
@eviltrout sure np, will revert.  Also note that the new bookmarks.json fixture is a bit of a hack from latest.json.  Might be good at some point to improve upon it if any tests get higher granularity.  It would be good to use your test accounts data.

**Comment by merefield** [comment]
@eviltrout something still not right?

**Comment by davidtaylorhq** [comment]
Yes it looks like there are prettier failures in 

```
app/assets/javascripts/discourse/app/controllers/preferences/interface.js
test/javascripts/acceptance/preferences-test.js
test/javascripts/helpers/site.js
```

The best thing here is to set up your editor to automatically format the code. There is some info on that [here](https://meta.discourse.org/t/prettier-code-formatting-tool/93212)

As for the feature itself, am I correct in saying it allows [/bookmarks](https://meta.discourse.org/bookmarks) as the homepage? We have the much nicer new [/my/activity/bookmarks](https://meta.discourse.org/my/activity/bookmarks) page, and I seem to remember some talk about deprecating/removing `/bookmarks`. @martin-brennan is that something you were thinking about?

**Comment by merefield** [comment]
Ooops apologies ... perhaps my prettier config is out.  I'll take a look.

**Comment by merefield** [comment]
@davidtaylorhq thanks for the pointers, that resolved it and I should be free of that issue for future PRs now I've got the config right!

**Comment by eviltrout** [comment]
This looks good now but I'd like to hear @martin-brennan's response before we merge.

**Comment by martin-brennan** [comment]
@davidtaylorhq thanks for pinging me, at this time we are leaving the /bookmarks page as is because it serves a slightly different purpose to the other bookmarks list. I think adding this functionality is fine for now.

File: test/javascripts/acceptance/preferences-test.js [hunk 1]
```diff
@@ -427,3 +427,43 @@ QUnit.test("can select an option from a dropdown", async (assert) => {
...
+    "it has the option to disable dark mode"
+  );
+});
+
+acceptance("User Preferences disabling dark mode", {
+  loggedIn: true,
+  settings: {
+    top_menu: "categories|latest|top|bookmarks",
+  },
+});
```
Comment (martin-brennan) [inline test/javascripts/acceptance/preferences-test.js:449]: I am not sure why all this dark mode stuff is here, possibly merge/rebase shenanigans?

**Review by martin-brennan — CHANGES_REQUESTED** [review]
This works great, thank you for adding specs as well. I think its fine the fixture follows latest because the /bookmarks list doesn't differ much from latest. If you just fix up the comment it should be good to merge.

Also in future could you please add a PR description of the what and why for the PR; I realise it is in Meta but having a succinct summary of what is changing helps a lot with reviews :+1:

File: test/javascripts/acceptance/preferences-test.js (hunk 1 above)
Comment (merefield) [reply test/javascripts/acceptance/preferences-test.js:449]: Good catch.  Yes a rebase issue.  Will address and revert.

Comment (merefield) [reply test/javascripts/acceptance/preferences-test.js]: @martin-brennan removed the imposter test and embellished the commit message to better describe the change.  thanks for your review!

**Review by eviltrout — APPROVED** [review]

**Pull request merged**